pulsar/producer_partition.go

```diff
@@ -481,10 +484,20 @@ func (p *partitionProducer) reconnectToBroker() {
 func (p *partitionProducer) runEventsLoop() {
 	for {
 		select {
-		case data := <-p.dataChan:
+		case data, ok := <-p.dataChan:
+			// when doClose() is call, p.dataChan will be closed, data will be nil
+			if !ok {
+				p.batchFlushTicker.Stop()
```

Comment:
p.batchFlushTicker will be stopped when closing. Why do we need to stop it here?